runtime/vm/resolvesupport.cpp

```diff
@@ -972,7 +972,30 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		
 			if (ramCPEntry != NULL) {
 				UDATA valueOffset = fieldOffset;
-
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				if ('Q' == *J9UTF8_DATA(signature)) {
+					J9FlattenedClassCache *flattenedClassCache = classFromCP->flattenedClassCache;
+					J9Class *flattenableClass = NULL;
+					UDATA index = 0;
+
+					Assert_VM_notNull(flattenedClassCache);
+					index = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
```

Comment:
Is there ever a case where you expect not to find the index? Wondering if this asserts could move into the callee.

Reply:
Still waiting for an answer here - if there is no case where you test the return value other than asserting it, the asserts may as well be in the callee.

Reply:
I moved the assertion to the callee, just forgot to remove it from the caller. I have made that change now.